File: docs/examples/5_search_courses.js
// Fetch API example (works in Node.js 18+ and browsers)
async function searchCourses() {
  try {
    const response = await fetch('http://localhost:3000/search', {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json',
      },
      
    });

    if (!response.ok) {
      throw new Error(`HTTP error! status: ${response.status}`);
    }

    const data = await response.json();
    console.log('Response:', data);
    return data;
  } catch (error) {
    console.error('Error:', error);
  }
}

// Usage
searchCourses();

File: docs/examples/5_search_courses.py
import requests
import json

def search_courses():
    url = "http://localhost:3000/search"
    
    data = None
    
    try:
        response = requests.get(url, json=data)
        response.raise_for_status()
        print("Response:", response.json())
        return response.json()
    except requests.exceptions.RequestException as e:
        print(f"Error: {e}")
        return None

# Usage
if __name__ == "__main__":
    search_courses()